### PR TITLE
Test add all auth

### LINKED ISSUE #2: Add tests for all /auth controllers
There are 4 controllers for `/auth`
- /register, /user, /login and /logout
- Provide a proper test suite for all of these endpoints

### PR DESCRIPTION
**Pull request by navjot-chahal**
### What this PR does:
- Adds mocha tests for `/auth/register`, `/auth/login`, `/auth/user` and `/auth/logout` routes.
- Changes logout request type to "POST" from "GET" for both server and client side.

### Screenshots / Videos:
- CLI mocha test results:
Screen Shot 2021-10-19 at 7 43 45 PM

### Any information needed to test this feature:
- Use `npm run dev` to start the test in the server folder.

closes #2

### CODE REVIEW

File: server/test/auth.test.js
Comment (ezeilo-su) [inline server/test/auth.test.js:85]: You can remove this comment as the code is clear what is does.

**Review by sundayezeilo — COMMENTED** [review]
Your tests look good! I Like those screenshots :+1:

File: client/src/helpers/APICalls/logout.ts
Comment (ugur-akin) [inline client/src/helpers/APICalls/logout.ts:6]: 🙌🏼

File: server/test/auth.test.js
Comment (ugur-akin) [inline server/test/auth.test.js:34]: `400` is probably not the most appropriate status code for this case (you can find some suggestions on what might be on this [thread](https://stackoverflow.com/questions/9269040/which-http-response-code-for-this-email-is-already-registered/9270432)). 

Ideally, we don't want to define success/failure of our test cases based on the implementation. Therefore I would recommend not looking at the implementation(s) of the feature(s) you're testing as much as possible while designing them (as if to pretend they aren't implemented yet) 👍🏼

File: server/test/auth.test.js
Comment (ugur-akin) [inline server/test/auth.test.js:80]: ❤️

**Review by ugur-akin — CHANGES_REQUESTED** [review]
Looks great, thanks for you work 👍🏼 I left a  minor comment, take a look and let's check back and merge this soon! 🎉

**Comment by rajivtitus** [comment]
@navjot-chahal why has this branch been merged without approval? :)

**Comment by navjot-chahal** [comment]
@rajivtitus There was a very minor change. Just the error code response is changed to `409` from `400`. Is there any issue with the last commit?

Also, if this merge was a violation then I will make sure to wait for approval for a minor change as well before merging.

**Comment by rajivtitus** [comment]
@navjot-chahal There are no issues with the last commit, good work here :)

In the future please wait for approval from a Hatchways team member prior to merging PRs. We review PRs once again to make sure the requested feedback has been implemented or any new issues have been introduced.

**Pull request merged**
